src/include/common/gate.h

```diff
@@ -33,11 +32,7 @@ class Gate {
    * Traverses the gate unless there are currently locks emplaced.  If there
    * are locks on the gate, spin until its free.
    */
-  void Traverse() {
```

Comment:
This is going to prevent inlining on what's ultimately a pretty short function on the txn begin critical path. We should discuss if we want this change and if there's another way to arrange the includes to make the <immintrin.h> less painful.

Reply:
One easy fix is to include emmintrin.h instead of immintrin.h, which is a much smaller header. I'll run it and find out the difference in time.